raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1543,9 +1516,40 @@ private long maybeAppendBatches(
         return timeUnitFlush;
     }
 
+    private long pollResigned(long currentTimeMs) throws IOException {
+        GracefulShutdown shutdown = this.shutdown.get();
+        ResignedState state = quorum.resignedStateOrThrow();
+
+        long endQuorumBackoffMs = maybeSendRequests(
```

Comment:
If the cluster resigns from candidateState, it will also send EndQuorumEpochRequest to all voters?